apps/weatherdashboard/weather_dashboard.star

```diff
@@ -0,0 +1,215 @@
+load("cache.star", "cache")
+load("encoding/json.star", "json")
+load("http.star", "http")
+load("render.star", "render")
+load("schema.star", "schema")
+
+AMBIENT_WEATHER_URL = "https://rt.ambientweather.net/v1/devices"
+
+def main(config):
+    current_data = get_weather_data(config)
+
+    outside_temp = current_data["tempf"]
+    feels_like = current_data["feelsLike"]
+    humidity = current_data["humidity"]
+
+    wind_spd_avg = current_data["windspdmph_avg10m"]
+    wind_dir = get_wind_direction(current_data["winddir"])
+
+    event_rain = current_data["eventrainin"]
+    weekly_rain = current_data["weeklyrainin"]
+
+    degree_sign = "°"
+
+    weather_info = []
+    weather_info.append(add_row("Temp", "{0}{1}".format(outside_temp, degree_sign)))
+    weather_info.append(add_row("Wind", "{0} {1}".format(wind_dir, wind_spd_avg)))
+    if config.bool("show_precip"):
+        weather_info.append(render.Box(width = 64, height = 1, color = config.get("divider_color", "#1167B1")))
+        weather_info.append(show_rainfall(weekly_rain, event_rain, config))
+    else:
+        weather_info.append(add_row("Feel", "{0}{1}".format(feels_like, degree_sign)))
+        weather_info.append(add_row("Humidity", humidity))
+
+    return render.Root(
+        child = render.Column(
+            expanded = True,
+            main_align = "space_around",
+            cross_align = "center",
+            children = weather_info,
+        ),
+    )
+
+def get_schema():
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.Text(
+                id = "application_key",
+                name = "Application Key",
+                desc = "Ambient Weather Application Key.",
+                icon = "key",
+            ),
+            schema.Text(
+                id = "api_key",
+                name = "API Key",
+                desc = "Ambient Weather API Key.",
+                icon = "key",
+            ),
+            schema.Toggle(
+                id = "show_precip",
+                name = "Show Rainfall",
+                desc = "Show rainfall totals",
+                icon = "compress",
+                default = True,
+            ),
+            schema.Color(
+                id = "divider_color",
+                name = "Divider Color",
+                desc = "The color of the dividers",
+                icon = "brush",
+                default = "#1167B1",
+            ),
+        ],
+    )
+
+# pragma mark Helper Methods
+
+def get_weather_data(config):
+    cached_data = cache.get("cached_data")
```

Comment:
Cache is scoped to the app, but shared among users. This cache key will need to incorporate something unique to the user, such as their API and application keys.

Reply:
Ah I didn't realize that, just updated it to use the api key to disambiguate